store/iavl/store.go

```diff
@@ -22,8 +22,34 @@ const (
 	defaultIAVLCacheSize = 10000
 )
 
-// LoadStore loads the iavl store
-func LoadStore(db dbm.DB, id types.CommitID, pruning types.PruningOptions, lazyLoading bool) (types.CommitStore, error) {
+var (
+	_ types.KVStore       = (*Store)(nil)
+	_ types.CommitStore   = (*Store)(nil)
+	_ types.CommitKVStore = (*Store)(nil)
+	_ types.Queryable     = (*Store)(nil)
+)
+
+// Store Implements types.KVStore and CommitKVStore.
+type Store struct {
+	tree Tree
+
+	// How many old versions we hold onto.
+	// A value of 0 means keep no recent states.
+	numRecent int64
+
+	// This is the distance between state-sync waypoint states to be stored.
+	// See https://github.com/tendermint/tendermint/issues/828
+	// A value of 1 means store every state.
+	// A value of 0 means store no waypoints. (node cannot assist in state-sync)
+	// By default this value should be set the same across all nodes,
+	// so that nodes can know the waypoints their peers store.
+	storeEvery int64
```

Comment:
Why not use a boolean here if we're only having two possible values?

Reply:
Not sure I follow? This was not introduced in this PR and relates to pruning. See [here](https://github.com/cosmos/cosmos-sdk/blob/v0.37.0/store/iavl/store.go#L111-L121) for pruning usage.

Reply:
I don't think this is supposed to be only two values. If the snapshotting frequency is every 10,000 blocks for state-sync, then a node can specify that they want to store every 10,000th block.

This will minimize storage requirements while still allowing node to participate in state-sync

Reply:
Got it, sorry I didn't realize this was just moved from somewhere else.